### ISSUE TITLE
elseOnSameLine unexpectedly wrapping else keyword despite using --elseposition same-line

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#782]
In the following example, the `elseOnSameLine` rule is unexpectedly wrapping the else keyword (even though we are using `--elseposition same-line`:

#### `swiftformat . --rules elseOnSameLine,indent --elseposition same-line`

```diff
struct Foo {

     func unexpectedElseWrap() {
         if let foo == bar {
             // ...
-        } else {
+        }
+        else {
             // ...
         }
     }

     func foo()
     {
         print()
     }

     func bar()
     {
         print()
     }

     func baaz()
     {
         print()
     }

     func quux()
     {
         print()
     }

     func quack()
     {
         print()
     }
 }
```

Interestingly, removing any of the `foo`/`bar`/`baaz`/`quux`/`quack` methods causes the rule to behave as expected.

**Comment by nicklockwood** [comment]
@calda I'm curious about the real-world case that lead to you discovering this, but FWIW it seems like it may not actually be a bug so much as a quirk:

The `elseOnSameLine` rule uses the `--allman` option in addition to `--elseposition` to determine its behavior. If you haven't set the `--allman` option explicitly, and you've disabled the `braces` rule then it tries to infer whether the file is using `--allman` braces by looking at the code in the file. Since all  the functions in this file are using allman braces, it assumes you want to use allman for the else statement as well.

**Comment by calda** [comment]
I also tried this with `--allman false` and am seeing the same result:
> `swiftformat . --rules elseOnSameLine,indent --elseposition same-line --allman false`

In our codebase we use `wrapMultilineStatementBraces` and the file in question is entirely comprised of multi-line function declarations (this looks similar to `--allman true` but is a tiny bit different).

```swift
func foo(
  bar: Bar,
  baaz: Baaz)
{
    print()
}
```

**Comment by nicklockwood** [comment]
Setting `--allman false` won't have any effect because it matches default (a limitation of the way I implement options, which I should probably fix).

It seems that the allman inference is flawed since it's detecting wrapped function declarations as allman when they're actually a special case. I'll see if I can fix that.

**Comment by calda** [comment]
Interesting! Good to know.

**Comment by nicklockwood** [comment]
@calda fixed in 0.47.3
